src/main/java/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package suppliers;
+
+import core.basesyntax.Colors;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE_COUNT);
+        Figure figure;
+        switch (index) {
+            case 1 :
+                figure = new Circle().setRandomCircle();
+                figure.getInfo();
+                break;
+            case 2 :
+                figure = new IsoscelesTrapezoid().setRandomIsoscelesTrapezoid();
+                figure.getInfo();
+                break;
+            case 3 :
+                figure = new Rectangle().setRandomRectangle();
+                figure.getInfo();
+                break;
+            case 4 :
+                figure = new RightTriangle().setRandomRightTriangle();
+                figure.getInfo();
+                break;
+            default:
+                figure = new Square().setRandomSquare();
+                figure.getInfo();
```

Comment:
lets move logic of generatiog of random figures from figure classes( Circle, Rectangle, ...) to this class (FigureSupplier). And lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)